### PR TITLE
fix for sync up job not working in 2.17 when upgraded from previous versions

### PR DESCRIPTION
**Pull request by rbhavna**
### Description
fix for sync up job not working in 2.17 when upgraded from previous versions

### Related Issues
Resolves https://github.com/opensearch-project/ml-commons/issues/3238

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java [hunk 1]
```diff
@@ -261,6 +261,7 @@ private void queryRunningModels(ActionListener<SearchResponse> listener) {
     private void triggerModelRedeploy(ModelAutoRedeployArrangement modelAutoRedeployArrangement) {
         if (modelAutoRedeployArrangement == null) {
             log.info("No more models in arrangement, skipping the redeployment");
+            redeployAModel();
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java:264]: Why should we redeploy the model if `modelAutoRedeployArrangement` is null?

Comment (rbhavna) [reply plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java:264]: This method calls for redeploying the next model. Not the same one. Ideally if this is null, [this](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java#L157) condition should have been satisfied and triggerModelRedeploy shouldn't be called. Alternatively we can call startCronjobAndClearListener() here as well.

**Review by ylwu-amzn — DISMISSED** [review]

File: plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java (hunk 1 above)
Comment (zane-neo) [reply plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java:264]: Please change this to `startCronjobAndClearListener` method call.

Comment (zane-neo) [reply plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java:264]: Also, can you add an else to this [if](https://github.com/opensearch-project/ml-commons/pull/3241/files#diff-a81f529712d8f8f4b41ad5f5a2e7c7cbf8712211f26d04c91a4b208c1c1436b3L168) to ensure the cronjob started in edge cases, e.g. user deleted all the models in the index.

**Review by ylwu-amzn — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Pull request merged**